### ISSUE TITLE
Browser compatibility should be clearly defined

### ISSUE THREAD

**Issue by karimsa** [jaydenwindle/decode2017#34]
Currently, the CSS generally assumes that the user is running a fairly up-to-date browser. Many merchants must cater to lesser browsers (IE 11 is still legacy).

Generally, some level of testing should be done via BrowserStack via Travis.

JS issue: pop-in `requestAnimationFrame` vendor prefixes + fallback (based on compatibility):

```javascript
window.requestAnimationFrame =
  window.requestAnimationFrame ||
  window.webkitRequestAnimationFrame ||
  window.mozRequestAnimationFrame ||
  window.msRequestAnimationFrame ||
  window.oRequestAnimationFrame ||
  function ( callback ) {
    setTimeout(callback, 1000 / 60)
  }
```

**Comment by jaydenwindle** [comment]
Good call. 

The easiest way to accomplish this would be to stay clear of features that don't have wide support yet. I don't think we really _need_ `requestAnimationFrame` or any of the `async` stuff that we have now to accomplish what we want to do. Plus the Shopify JS API uses the classic callback style, so I think we'll be fine to stick with that.

**Comment by karimsa** [comment]
"wide support" is a very loosely defined term. What we need to do is decide a definite list of browsers & versions that we'd like to support. Preferably using a [`browserslist`](https://www.npmjs.com/package/browserslist) query as that is well defined yet dynamic.

`requestAnimationFrame` is incredibly important for data binding and has fairly strong support from all major browsers. Dropping in a 5 line polyfill is not a big deal and is only needed depending on what older browsers we are adding support for.

There is no more ES2015 features being used in the JS, but ideally, there should be a babel setup with babel-preset-env so we can use ESNext features.

The SCSS should ideally be swapped out for postcss and we can pop-in autoprefixer and plugins like that for easier browser support.

**Comment by jaydenwindle** [comment]
True, how about we define support the same as Bootstrap 4 does? [IE10+, Chrome, Firefox, Edge, Opera](https://v4-alpha.getbootstrap.com/getting-started/browsers-devices/#desktop-browsers)

As for data binding, I think it would be easiest to drop in a framework like vue.js to handle that as opposed to coding it from scratch. I just don't like reinventing wheels.

Also having autoprefixer is a great idea, but I really don't think it's necessary to get rid of sass in favor of postcss. Personally I love sass syntax, and it's really closely tied with the ruby ecosystem, not to mention that it's plug and play with the rails asset pipeline. Perhaps we could use [autoprefixer-rails](https://github.com/ai/autoprefixer-rails)?
